test/priority_boost_integrationtest.cc

```diff
@@ -0,0 +1,190 @@
+ public:
+  struct SystemSchedInfo {
+    std::string comm;
+    int prio;
+    int nice;
+    int rt_prio;
+    int policy;
+
+    std::string ToString() const {
+      std::ostringstream oss;
+      oss << "SystemSchedInfo{";
+      oss << "comm: " << comm << ", prio: " << prio << ", nice: " << nice
+          << ", rt_prio: " << rt_prio << ", policy: " << policy;
+      oss << "}";
+      return oss.str();
+    }
+  };
+
+  static std::optional<SystemSchedInfo> GetSchedInfo(int tid) {
+    base::StackString<128> stat_path("/proc/%d/stat", tid);
+    std::string line;
+    if (!base::ReadFile(stat_path.c_str(), &line)) {
+      PERFETTO_ELOG("Can't read file: %s", stat_path.c_str());
+      return std::nullopt;
+    }
+    std::vector parts = base::SplitString(line, " ");
+    int prio = base::StringToInt32(parts[17]).value();
+    int nice = base::StringToInt32(parts[18]).value();
+    int rt_prio = base::StringToInt32(parts[39]).value();
+    int policy = base::StringToInt32(parts[40]).value();
+    SystemSchedInfo info{parts[1], prio, nice, rt_prio, policy};
+    return info;
+  }
+};
+}  // namespace
+
+TEST_F(PerfettoPriorityBoostIntegrationTest, TestTraced) {
+  base::TestTaskRunner task_runner;
+
+  TestHelper helper(&task_runner);
+  helper.StartServiceIfRequired();
+
+  int traced_tid = -1;
+  int traced_probes_tid = -1;
+
+#if PERFETTO_BUILDFLAG(PERFETTO_START_DAEMONS)
+  PERFETTO_DLOG("PERFETTO_START_DAEMONS");
+  ProbesProducerThread probes(GetTestProducerSockName());
+  probes.Connect();
+#elif
+  GTEST_SKIP() << "This test requires PERFETTO_START_DAEMONS or ANDROID_BUILD";
+  PERFETTO_DLOG("NOT PERFETTO_START_DAEMONS");
+#endif
+
+  helper.ConnectConsumer();
+  helper.WaitForConsumerConnect();
+  helper.WaitForDataSourceConnected("linux.ftrace");
+
+  TraceConfig trace_config;
+  trace_config.add_buffers()->set_size_kb(64);
+  auto* priority_boost_config = trace_config.mutable_priority_boost();
+  priority_boost_config->set_policy(
+      protos::gen::PriorityBoostConfig::POLICY_SCHED_FIFO);
+  priority_boost_config->set_priority(42);
+
+  auto* ds_config = trace_config.add_data_sources()->mutable_config();
+  ds_config->set_name("linux.ftrace");
+  ds_config->set_target_buffer(0);
+
+  auto* ds_priority_boost_config = ds_config->mutable_priority_boost();
+  ds_priority_boost_config->set_policy(
+      protos::gen::PriorityBoostConfig::POLICY_SCHED_OTHER);
+  ds_priority_boost_config->set_priority(7);
+
+  protos::gen::FtraceConfig ftrace_config;
+  ftrace_config.add_ftrace_events("sched_switch");
+  ds_config->set_ftrace_config_raw(ftrace_config.SerializeAsString());
+
+  helper.StartTracing(trace_config);
+  helper.WaitForDataSourceConnected("linux.ftrace");
+
+#if PERFETTO_BUILDFLAG(PERFETTO_ANDROID_BUILD)
+  PERFETTO_DLOG("PERFETTO_ANDROID_BUILD");
+  traced_tid = PidForProcessName("/system/bin/traced");
+  traced_probes_tid = PidForProcessName("/system/bin/traced_probes");
```

Comment:
rather than hardcoding the paths could we get them from QueryServiceState? I think that works for both cases